### PR TITLE
[desktop-lite] - vnc_resolution - set thru env var

### PR DESCRIPTION
**Pull request by gauravsaini04**
**Feature Name**

* Desktop-lite

**Description**

* Escaped the variables in `desktop-init.sh` file at the init of the file as a solution to feature's issue [#945](https://github.com/devcontainers/features/issues/945)

**_changelog_**

* VNC_RESOLUTION as an env. var. should get picked up by the `desktop-init.sh` file due to the above mentioned changes

### CODE REVIEW

File: src/desktop-lite/install.sh [hunk 1]
```diff
@@ -300,11 +300,11 @@ user_name="${USERNAME}"
 group_name="$(id -gn ${USERNAME})"
 LOG=/tmp/container-init.log
 
-export DBUS_SESSION_BUS_ADDRESS="${DBUS_SESSION_BUS_ADDRESS:-"autolaunch:"}"
-export DISPLAY="${DISPLAY:-:1}"
-export VNC_RESOLUTION="${VNC_RESOLUTION:-1440x768x16}" 
-export LANG="${LANG:-"en_US.UTF-8"}"
-export LANGUAGE="${LANGUAGE:-"en_US.UTF-8"}"
+export DBUS_SESSION_BUS_ADDRESS="\${DBUS_SESSION_BUS_ADDRESS:-"autolaunch:"}"
```
[the remark this answers is not in the document]
Comment (gauravsaini04) [reply src/desktop-lite/install.sh:303]: increased the patch version for the feature

File: src/desktop-lite/install.sh [hunk 2]
```diff
@@ -329,10 +329,18 @@ keepRunningInBackground()
     (\$2 bash -c "while :; do echo [\\\$(date)] Process started.; \$3; echo [\\\$(date)] Process exited!; sleep 5; done 2>&1" | sudoIf tee -a /tmp/\$1.log > /dev/null & echo "\$!" | sudoIf tee /tmp/\$1.pid > /dev/null)
 }
 
+check_packages() {
```
[the remark this answers is not in the document]
Comment (gauravsaini04) [reply src/desktop-lite/install.sh:332]: yes it's required as `sudo: command not found` was showing up during build of the feature

File: src/desktop-lite/install.sh [hunk 3]
```diff
@@ -329,10 +329,18 @@ keepRunningInBackground()
...
+        apt_get_update
+        apt-get -y install --no-install-recommends "\$@"
+    fi
+}
+
 # Use sudo to run as root when required
 sudoIf()
 {
     if [ "\$(id -u)" -ne 0 ]; then
+        check_packages sudo
```
[the remark this answers is not in the document]
Comment (gauravsaini04) [reply src/desktop-lite/install.sh:343]: not the test case but build was failing for the feature

File: test/desktop-lite/scenarios.json [hunk 4]
```diff
@@ -4,5 +4,22 @@
...
             "desktop-lite": {}
         }
+    },
+    "test_vnc_resolution_as_env_var": {
+        "image": "ubuntu:noble",
+        "features": {
+            "desktop-lite": {}
+        }
+        ,
+        "containerEnv": {
```
[the remark this answers is not in the document]
Comment (gauravsaini04) [reply test/desktop-lite/scenarios.json:14]: remoteEnv scenarios wasn't able to change the value of VNC_RESOLUTION through the env. var. method while the entrypoint script loaded and ran, pls check the test run for it

File: src/desktop-lite/install.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (gauravsaini04) [reply src/desktop-lite/install.sh]: found in a build of a container but when I now removed it, worked without any issues.

File: test/desktop-lite/test_vnc_resolution_as_container_env_var.sh [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    PID=$(lsof -i :$port | awk 'NR==2 {print $2}')
+    GREEN='\033[0;32m'; NC='\033[0m'; RED='\033[0;31m'; YELLOW='\033[0;33m';
+    if [ -n "$PID" ]; then
+        CMD=$(ps -p $PID -o cmd --no-headers)
+        echo -e "${GREEN}Command running on port $port: ${YELLOW}$CMD${NC}"
+    else
+        echo -e "${RED}No process found listening on port $port.${NC}"
+        exit 1
+    fi
+}
```
Comment (samruddhikhandale) [inline test/desktop-lite/test_vnc_resolution_as_container_env_var.sh:21]: Is it possible to add a check to validate `VNC_RESOLUTION` variable value to matches with what we passed?

Comment (gauravsaini04) [reply test/desktop-lite/test_vnc_resolution_as_container_env_var.sh:22]: added

**Review by samruddhikhandale — APPROVED** [review]
Thanks!

**Pull request merged**